tests/test_backend.py

```diff
@@ -600,30 +601,165 @@ def test_persist(self, cache: RedisCache):
         assert ttl is None
         assert cache.persist("not-existent-key") is False
 
-    def test_expire(self, cache: RedisCache):
-        cache.set("foo", "bar", timeout=None)
-        assert cache.expire("foo", 20) is True
-        ttl = cache.ttl("foo")
-        assert pytest.approx(ttl) == 20
-        assert cache.expire("not-existent-key", 20) is False
+    @pytest.mark.parametrize(
+        "initial_timeout, new_timeout, params, expected_result, expected_ttl",
+        [
+            # Basic expire functionality (existing test case)
+            (None, 20, {}, True, 20),
+            # NX tests (only set if key has no expiry)
+            # Should work - key has no expiry
+            (None, 30, {"nx": True}, True, 30),
+            # Should fail - key already has expiry
+            (20, 30, {"nx": True}, False, 20),
+            # XX tests (only set if key has existing expiry)
+            # Should work - key has expiry
+            (20, 30, {"xx": True}, True, 30),
+            # Should fail - key has no expiry
+            (None, 30, {"xx": True}, False, None),
+            # GT tests (only set if new expiry is greater than current)
+            # Should work - new timeout > current
+            (20, 30, {"gt": True}, True, 30),
+            # Should fail - new timeout < current
+            (30, 20, {"gt": True}, False, 30),
+            # LT tests (only set if new expiry is less than current)
+            # Should work - new timeout < current
+            (30, 20, {"lt": True}, True, 20),
+            # Should fail - new timeout > current
+            (20, 30, {"lt": True}, False, 20),
+        ],
+    )
+    def test_expire_with_conditions(
+        self,
+        cache: RedisCache,
+        initial_timeout,
+        new_timeout,
+        params,
+        expected_result,
+        expected_ttl,
+    ):
+        cache.set("foo", "bar", timeout=initial_timeout)
+        result = cache.expire("foo", new_timeout, **params)
+
+        assert result is expected_result
+        if expected_ttl is not None:
+            assert pytest.approx(cache.ttl("foo")) == expected_ttl
```

Comment:
you need to add a higher tolerance have a look at the [docs](https://docs.pytest.org/en/latest/reference/reference.html#pytest.approx)